spec/rspec/matchers/built_in/be_spec.rb

```diff
@@ -85,6 +85,21 @@
         expect(actual).to be_happy
       }.to fail_with("expected `#{actual.inspect}.happy?` to be truthy, got false")
     end
+
+    it "does not allow non-supplied dynamic matchers to pass on spies" do
+      thing = spy("thing", has_recv?: true)
+
+      expect(thing).to have_recv(:foo)
+      expect(thing).to have_received(:has_recv?).with(:foo)
+
+      begin
```

Comment:
Can we say that this has two checks for the same thing that won’t fail separately?
One is that the method wasn’t called on the spy, and another that the expectation failed?

What if we just

expect {
         expect(thing).to have_receivex(:foo)
         }.to fail_with(/some message/)

I recall this is pretty common in our specs

Reply:
If I understand the question, then yes - the second assertion in each of the updated tests (did the spy register receiving the method) _could_ be tested separately, if that's preferred.

I've always had a little trouble deciding where the boundary should be for testing actions that have multiple related effects like this, but since the first part of the spec is probably the simplest way to set up the second part of the spec in each case, I think one of the new tests in each case would be identical to the current test, and the other would be just the first segment of each? My threshold for splitting in those cases has (lately) just been "can I phrase the test description cleanly enough to be clear with them together?".

In some of my projects I use a `rescued { do_the_thing }` spec helper method, if we have some analogue to that (or would like one) it would make the split a little cleaner, though with the fail-matcher it's probably less useful than usual.

Up to you, happy to work it however :-)